### PR TITLE
add vapor toolbox

### PR DESCRIPTION
**Pull request by tanner0101**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Adds the CLI tool and dependencies for the [Vapor web framework](https://github.com/vapor/vapor). Current [docs](https://docs.vapor.codes/2.0/getting-started/toolbox/).

### CODE REVIEW

File: Formula/vapor.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Vapor < Formula
+  desc "Simplifies common command-line tasks when using Vapor."
+  homepage "https://vapor.codes"
+  url "https://github.com/vapor/toolbox/releases/download/3.0.3/macOS-sierra.tar.gz"
```
Comment (DomT4) [inline Formula/vapor.rb]: This will need to be built from source.

File: Formula/vapor.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  version "3.0.3"
+  sha256 "0461af9672896918e66bf3f698f0e78df0b8eb51c2a4a115919a0a30e055452c"
+
+  depends_on "vapor/tap/ctls" => :run
```
Comment (DomT4) [inline Formula/vapor.rb]: ... And you can't have mandatory cross-platform dependencies.

Comment (DomT4) [reply Formula/vapor.rb]: Er, cross-tap. Not cross-platform. Maybe I should sleep 😅.

**Comment by ilovezfs** [comment]
Thanks for the pull request @tanner0101! Feel free to open a fresh pull request when you have the software building from source, there are no cross-tap dependencies, and the formula has a `test do` block that exercises and verifies the functionality of the software.

**Comment by tanner0101** [comment]
Ah ok, the no cross-tap thing is a pain. I'll wait until we don't need that anymore to resubmit. As for building from source, does Homebrew take care of caching that automatically then? Can't imagine something like `brew install mysql` compiles from source haha.

**Comment by DomT4** [comment]
Homebrew provides precompiled binaries (`bottles`) for the three most recent stable major macOS versions, at the moment High Sierra, Sierra & El Capitan. Given how few users don't use those, the practical result is that >90% of people will never build any given formula from source.

**Pull request closed without merging**